### PR TITLE
empty string are invalid emails and should considered so and not thro…

### PR DESCRIPTION
**Pull request by norkans7**
…w error

### CODE REVIEW

File: temba/utils/tests.py [hunk 1]
```diff
@@ -322,8 +322,9 @@ def test_is_valid_address(self):
         self.assertFalse(is_valid_address("a@b"))
         self.assertFalse(is_valid_address(" @ .c"))
         self.assertFalse(is_valid_address("a @b.c"))
+        self.assertFalse(is_valid_address("{@flow.email}"))
         self.assertTrue(is_valid_address("a@b.c"))
-        self.assertTrue(is_valid_address('"Abc@def"+label@example.com'))
+        self.assertTrue(is_valid_address('Abc@def"+label@example.com'))
```
Comment (norkans7) [inline temba/utils/tests.py]: I just considered valid email do not start with ", thoughts?

File: temba/flows/tests.py [hunk 2]
```diff
@@ -2733,7 +2733,7 @@ def test_email_action(self):
         self.assertEqual(logs[0].level, ActionLog.LEVEL_INFO)
         self.assertEqual(logs[0].text, "&quot;Test Contact uses phone (206) 555-0100&quot; would be sent to testcontact@nyaruka.com")
         self.assertEqual(logs[1].level, ActionLog.LEVEL_WARN)
-        self.assertEqual(logs[1].text, "Some email address appear to be invalid: Test Contact, xyz")
+        self.assertEqual(logs[1].text, "Some email address appear to be invalid: Test Contact, xyz, ")
```
Comment (nicpottier) [inline temba/flows/tests.py]: Maybe we should put quotes around these so that the error messages is clearer?

File: temba/utils/tests.py (hunk 1 above)
Comment (nicpottier) [inline temba/utils/tests.py]: Seems according to: https://en.wikipedia.org/wiki/Email_address that should be safe

Comment (norkans7) [inline temba/utils/tests.py]: then ```{@flow.email}``` is a valid email? 

I was trying to make that one never considered as valid

**Comment by norkans7** [comment]
@nicpottier remove the email validation it seems impossible to have a regex that will not match ```{@flow.email}```

**Comment by nicpottier** [comment]
@norkans7 Not sure I understand, what is {@flow.email}? Is that a case we see? Wouldn't it be @flow.email or @{flow.email}?

**Comment by norkans7** [comment]
it all started with https://sentry.io/nyaruka/textit/issues/157650498/events/4663358415/

**Comment by nicpottier** [comment]
Ok, ya, so that's invalid. While `{` is a valid "local" part of an email address, `}` can't be in a hostname, so we should be throwing those out. 

What about using a library like: 
  https://pypi.python.org/pypi/validate_email

File: temba/utils/email.py [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
-ADDRESS_REGEX = regex.compile(r'\S+@\S+\.\S+')
+from validate_email import validate_email
 
 
 def is_valid_address(address):
     """
     Very loose email address check
     """
-    return address and ADDRESS_REGEX.match(address)
+    return address and validate_email(address, check_mx=True)
```
Comment (nicpottier) [inline temba/utils/email.py]: Let's not check the MX, that actually does a call out to the server and it isn't reliable. (some servers don't accept connections from any IP) I think we should be able to remove pyDNS in that case as well.

File: temba/utils/email.py [hunk 4]
```diff
@@ -1,20 +1,41 @@
...
 
+from django.core.exceptions import ValidationError
 from django.core.mail import EmailMultiAlternatives, send_mail
+from django.core.validators import EmailValidator
 from django.template import loader, Context
 from django.conf import settings
 
 
-ADDRESS_REGEX = regex.compile(r'\S+@\S+\.\S+')
+class TembaEmailValidator(EmailValidator):
```
Comment (norkans7) [inline temba/utils/email.py]:
```python
class EmailValidator(object):
    ...
    user_regex = re.compile(
        r"(^[-!#$%&'*+/=?^_`{}|~0-9A-Z]+(\.[-!#$%&'*+/=?^_`{}|~0-9A-Z]+)*\Z"  # dot-atom
        r'|^"([\001-\010\013\014\016-\037!#-\[\]-\177]|\\[\001-\011\013\014\016-\177])*"\Z)',  # quoted-string
        re.IGNORECASE)
    domain_regex = re.compile(
        # max length of the domain is 249: 254 (max email length) minus one
        # period, two characters for the TLD, @ sign, & one character before @.
        r'(?:[A-Z0-9](?:[A-Z0-9-]{0,247}[A-Z0-9])?\.)+(?:[A-Z]{2,6}|[A-Z0-9-]{2,}(?<!-))\Z',
        re.IGNORECASE)

      ...
```

Comment (norkans7) [inline temba/utils/email.py]: domain_regex change the max length

user_regex includes now 
```python 
\040 # space
\042 # "
```
those were skipped

**Pull request merged**